src/store-api/src/region_request.rs

```diff
@@ -334,165 +329,30 @@ fn make_region_truncate(truncate: TruncateRequest) -> Result<Vec<(RegionId, Regi
-#[allow(unused)]
 fn make_region_bulk_inserts(request: BulkInsertRequest) -> Result<Vec<(RegionId, RegionRequest)>> {
     let Some(Body::ArrowIpc(request)) = request.body else {
         return Ok(vec![]);
     };
 
-    let mut region_requests: HashMap<u64, BulkInsertPayload> =
-        HashMap::with_capacity(request.region_selection.len());
-
-    let schema_data = FlightData::decode(request.schema.clone()).context(ProstSnafu)?;
-    let payload_data = FlightData::decode(request.payload.clone()).context(ProstSnafu)?;
-    let mut decoder = FlightDecoder::default();
-    let _schema_message = decoder.try_decode(schema_data).context(FlightCodecSnafu)?;
-    let FlightMessage::Recordbatch(rb) =
-        decoder.try_decode(payload_data).context(FlightCodecSnafu)?
-    else {
-        unreachable!("Always expect record batch message after schema");
-    };
-
-    for region_selection in request.region_selection {
-        let region_id = region_selection.region_id;
-        let region_mask = BooleanArray::new(
-            BooleanBuffer::new(Buffer::from(region_selection.selection), 0, rb.num_rows()),
-            None,
-        );
-
-        let region_batch = if region_mask.true_count() == rb.num_rows() {
-            rb.df_record_batch().clone()
-        } else {
-            arrow::compute::filter_record_batch(rb.df_record_batch(), &region_mask)
-                .context(DecodeArrowIpcSnafu)?
-        };
-
-        region_requests.insert(region_id, BulkInsertPayload::ArrowIpc(region_batch));
-    }
-
-    let result = region_requests
-        .into_iter()
-        .map(|(region_id, payload)| {
-            (
-                region_id.into(),
-                RegionRequest::BulkInserts(RegionBulkInsertsRequest {
-                    region_id: region_id.into(),
-                    payloads: vec![payload],
-                }),
-            )
-        })
-        .collect::<Vec<_>>();
-    Ok(result)
-}
-
-/// Convert [BulkInsertRequest] to [RegionRequest] and group by [RegionId].
-fn make_region_rows_bulk_inserts(
-    request: BulkInsertRequest,
-) -> Result<Vec<(RegionId, RegionRequest)>> {
-    let Some(Body::ArrowIpc(request)) = request.body else {
-        return Ok(vec![]);
-    };
-
-    let mut region_requests: HashMap<u64, BulkInsertPayload> =
-        HashMap::with_capacity(request.region_selection.len());
-
-    let decode_timer = metrics::CONVERT_REGION_BULK_REQUEST
+    let decoder_timer = metrics::CONVERT_REGION_BULK_REQUEST
         .with_label_values(&["decode"])
         .start_timer();
     let schema_data = FlightData::decode(request.schema.clone()).context(ProstSnafu)?;
     let payload_data = FlightData::decode(request.payload.clone()).context(ProstSnafu)?;
     let mut decoder = FlightDecoder::default();
-    let _schema_message = decoder.try_decode(schema_data).context(FlightCodecSnafu)?;
+    let _ = decoder.try_decode(schema_data).context(FlightCodecSnafu)?;
```

Comment:
So why do we need the schema_data?

Reply:
> So why do we need the schema_data?

`FlightDecode` needs this function call to fill the `FlightDecode::schema` field:
https://github.com/GreptimeTeam/greptimedb/blob/aab7451aa23d3e141fea0ee3a8b9478e82f15cdb/src/common/grpc/src/flight.rs#L164

Otherwise it will raise an error here: https://github.com/GreptimeTeam/greptimedb/blob/aab7451aa23d3e141fea0ee3a8b9478e82f15cdb/src/common/grpc/src/flight.rs#L169